### PR TITLE
UI: Show parsed certificate data in PKI

### PR DESCRIPTION
**Pull request by hashishaw**
This PR sets the parameters parsed from a certificate (on an issuer or certificate, and after many actions) to its own attribute which is then rendered as a separate component. For these fields only, the fields only show when there is a value present.

### CODE REVIEW

File: ui/app/serializers/pki/issuer.js [hunk 1]
```diff
@@ -23,10 +23,9 @@ export default class PkiIssuerSerializer extends ApplicationSerializer {
   normalizeResponse(store, primaryModelClass, payload, id, requestType) {
     if (payload.data.certificate) {
       // Parse certificate back from the API and add to payload
-      const parsedCert = parseCertificate(payload.data.certificate);
-      const data = { issuer_ref: payload.issuer_id, ...payload.data, ...parsedCert };
-      const json = super.normalizeResponse(store, primaryModelClass, { ...payload, data }, id, requestType);
-      return json;
+      const parsedCert = parseCertificate(payload.data.certificate || payload.data.csr);
```
Comment (hellobontempo) [inline ui/app/serializers/pki/issuer.js:26]: since csr's are technically not certificates I don't think they'll be parsed, if I remember correctly, they hit the error block and return `can_parse: false`

File: ui/lib/pki/addon/components/parsed-certificate-info-rows.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import Component from '@glimmer/component';
+
+/**
+ * Expects to have parsedCertificate from parse-pki-cert util passed in, and will render attributes if values exist
+ */
+export default class ParsedCertificateInfoRowsComponent extends Component {
+  get possibleFields() {
+    return [
+      { key: 'serial_number' },
```
[the remark this answers is not in the document]
Comment (hashishaw) [reply ui/lib/pki/addon/components/parsed-certificate-info-rows.js:9]: I think it we could potentially export the keys expected from the parser from that util, and import them here. 
I was basing these keys off the [typscript definition here](https://github.com/hashicorp/vault/blob/0496c6107b5f5079b09c0f104c341ad8b4d8768c/ui/types/vault/utils/parse-pki-cert.d.ts), so it seems like that needs to be updated to match the code linked above?

[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/lib/pki/addon/components/parsed-certificate-info-rows.js]: Yeah, that makes sense! I totally forgot about the ts file. 😓

File: ui/lib/pki/addon/components/page/pki-issuer-rotate-root.hbs [hunk 3]
```diff
@@ -111,6 +111,7 @@
 {{#if (eq this.displayedForm "use-old-settings")}}
   {{#if @newRootModel.id}}
     <PkiInfoTableRows @model={{@newRootModel}} @displayFields={{this.displayFields}} />
+    <ParsedCertificateInfoRows @model={{@newRootModel.parsedCertificate}} />
```
Comment (hashishaw) [inline ui/lib/pki/addon/components/page/pki-issuer-rotate-root.hbs:114]: In other other templates I added `parsedCertificates` to the fieldNames array and checked for it within the loop. Here I just added it to the template after the other fields are rendered. Is there a preference between the two approaches?

File: ui/app/models/pki/action.js [hunk 4]
```diff
@@ -26,6 +26,15 @@ const validations = {
       message: `Issuer name must be unique across all issuers and not be the reserved value 'default'.`,
     },
   ],
+  keyName: [
```
Comment (hashishaw) [inline ui/app/models/pki/action.js:29]: Screenshot 2023-04-11 at 9 45 48 AM

File: ui/app/models/pki/issuer.js [hunk 5]
```diff
@@ -19,21 +19,7 @@ const inputFields = [
...
-      'subjectSerialNumber',
-      'keyId',
-      'altNames',
-      'uriSans',
-      'ipSans',
-      'otherSans',
-      'notValidBefore',
-      'notValidAfter',
-    ],
+    default: ['certificate', 'caChain', 'commonName', 'issuerName', 'issuerId', 'keyId', 'parsedCertificate'],
```
Comment (Monkeychip) [inline ui/app/models/pki/issuer.js:22]: 😎

File: ui/app/serializers/pki/action.js [hunk 6]
```diff
@@ -6,14 +6,29 @@
...
   };
 
+  normalizeResponse(store, primaryModelClass, payload, id, requestType) {
+    if (payload.data.certificate) {
+      // Parse certificate back from the API and add to payload
+      const parsedCert = parseCertificate(payload.data.certificate);
+      const data = {
+        ...payload.data,
+        common_name: parsedCert.common_name,
+        ...parsedCert,
```
Comment (hellobontempo) [inline ui/app/serializers/pki/action.js:24]: Do we still need to spread the parsed attrs here?

File: ui/app/utils/parse-pki-cert-oids.js [hunk 7]
```diff
@@ -2,7 +2,7 @@
  * Copyright (c) HashiCorp, Inc.
  * SPDX-License-Identifier: MPL-2.0
  */
-import camelizeKeys from 'vault/utils/camelize-object-keys';
```
Comment (Monkeychip) [inline ui/app/utils/parse-pki-cert-oids.js:5]: I can't even find this under utils... ?? But nice replacement.

File: ui/lib/pki/addon/components/page/pki-issuer-rotate-root.hbs (hunk 3 above)
Comment (Monkeychip) [reply ui/lib/pki/addon/components/page/pki-issuer-rotate-root.hbs:114]: I _think_ my preference is to have it in the template (a little easier to read at first glance). However, it's not a strong preference, I could go either way.

File: ui/lib/pki/addon/components/parsed-certificate-info-rows.hbs
Comment (Monkeychip) [inline ui/lib/pki/addon/components/parsed-certificate-info-rows.hbs:1]: any chance the `this.possibleFields` array is empty and this doesn't render anything? If it is possible should we display something?

File: ui/lib/pki/addon/components/page/pki-issuer-rotate-root.hbs (hunk 3 above)
Comment (hellobontempo) [reply ui/lib/pki/addon/components/page/pki-issuer-rotate-root.hbs:114]: I think template looping can be hard to follow sometimes. Especially since sometimes we're looping through the object of expanded attributes, and sometimes the array of field names. 

I like that this makes it easy to track down where the parsed data that's displayed is coming from, and we can consistently display it at the bottom without having to worry about the fieldName order

File: ui/lib/pki/addon/components/parsed-certificate-info-rows.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import Component from '@glimmer/component';
+import { parsedParameterKeys } from 'vault/utils/parse-pki-cert-oids';
+
```
Comment (Monkeychip) [inline ui/lib/pki/addon/components/parsed-certificate-info-rows.js:3]: Can we add documentation here?

Comment (Monkeychip) [reply ui/lib/pki/addon/components/parsed-certificate-info-rows.js:3]: If it's not standard practice in the PKI addon components, then ignore.

File: ui/lib/pki/addon/components/parsed-certificate-info-rows.hbs [hunk 9]
```diff
@@ -0,0 +1,9 @@
+{{#each this.possibleFields as |field|}}
+  {{#let (get @model field.key) as |value|}}
+    <InfoTableRow
+      @label={{or field.label (humanize (dasherize field.key))}}
```
Comment (hellobontempo) [inline ui/lib/pki/addon/components/parsed-certificate-info-rows.hbs:4]: 👏

File: ui/lib/pki/addon/components/page/pki-issuer-rotate-root.hbs (hunk 3 above)
Comment (hashishaw) [reply ui/lib/pki/addon/components/page/pki-issuer-rotate-root.hbs:114]: Great, thanks for the feedback! I'll add it consistently to the template only 👍

File: ui/lib/pki/addon/components/parsed-certificate-info-rows.js (hunk 8 above)
Comment (hashishaw) [reply ui/lib/pki/addon/components/parsed-certificate-info-rows.js:3]: Sure, I can add the JSDoc formatted notes

**Review by Monkeychip — APPROVED** [review]
Nothing blocking from me. But maybe wait for the ✅ from Claire who is also reviewing.

File: ui/app/utils/parse-pki-cert-oids.js (hunk 7 above)
Comment (hellobontempo) [reply ui/app/utils/parse-pki-cert-oids.js:5]: It's a new one I added!

File: ui/lib/pki/addon/components/parsed-certificate-info-rows.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+ * @module ParsedCertificateInfoRowsComponent
+ * Renders attributes parsed from a PKI certificate (provided from parse-pki-cert util). It will only render rows for
+ * defined values that match `parsedParameterKeys` imported from the helper. It never renders common_name, even though
+ * the value is returned from the parse cert util, because `common_name` is important to PKI and we render it at the top.
+ *
+ * @example ```js
+ * <ParsedCertificateInfoRows @parsedCertificate={{@model.parsedCertificate}} />
+ * ```
+ *
+ * @param {object} parsedCertificate - object of parsed attributes from parse-pki-cert util
```
Comment (hellobontempo) [inline ui/lib/pki/addon/components/parsed-certificate-info-rows.js:14]: It looks like the example above and here this arg should be `@model` based on the template?
```suggestion
 * @param {object} model - object of parsed attributes from parse-pki-cert util
```

**Review by hellobontempo — APPROVED** [review]
Thank you for tackling this! Really nice improvement to the PKI work. Just a small doc update comment then 🚢

File: ui/lib/pki/addon/components/parsed-certificate-info-rows.js (hunk 10 above)
Comment (hashishaw) [reply ui/lib/pki/addon/components/parsed-certificate-info-rows.js:14]: Ah thank you! Spent too long looking at this 😵

File: ui/app/serializers/pki/issuer.js [hunk 11]
```diff
@@ -4,29 +4,31 @@
...
-  constructor() {
-    super(...arguments);
-    // remove following attrs from serialization
-    const attrs = ['caChain', 'certificate', 'issuerId', 'keyId', ...parsedParameters];
-    this.attrs = attrs.reduce((attrObj, attr) => {
-      attrObj[attr] = { serialize: false };
-      return attrObj;
-    }, {});
-  }
+  attrs = {
```
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/serializers/pki/issuer.js:12]: But you're using it still! the `serialize: false` one!
